planner/core/task.go

```diff
@@ -1035,11 +1033,15 @@ func (t *copTask) convertToRootTaskImpl(ctx sessionctx.Context) *rootTask {
 		}
 		ts := tp.(*PhysicalTableScan)
 		prevColumnLen := len(ts.Columns)
-		prevSchema = ts.schema.Clone()
+		prevSchema := ts.schema.Clone()
 		ts.Columns = ExpandVirtualColumn(ts.Columns, ts.schema, ts.Table.Columns)
 		if len(ts.Columns) > prevColumnLen {
```

Comment:
Can we directly change the condition to `t.needExtraProj == false || len(ts.Columns) > prevColumnLen`?